### ISSUE TITLE
SQL tests take a long time to run

### ISSUE THREAD

**Issue by roaks3** [hashicorp/terraform-provider-google#13603]
# Background

Our SQL tests each take a long time to run, in some cases >30 minutes (we had to bump the timeout recently to 40 minutes [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/7118)). This has implications for developers/contributors running tests locally, VCR tests run in PRs, and our nightly tests.

The worst offenders from a recent nightly run:
- 3181s TestAccSqlDatabaseInstance_backupUpdate
- 2988s TestAccSqlDatabaseInstance_createFromBackup
- 2293s TestAccSqlDatabaseInstance_deleteDefaultUserBeforeSubsequentApiCalls
- 2213s TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRangeReplica
- 2209s TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled
- 2184s TestAccSqlDatabaseInstance_mysqlMajorVersionUpgrade
- 2083s TestAccSqlDatabaseInstance_slave
- 1955s TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRange
- 1809s TestAccSqlDatabaseInstance_highAvailability

We would like to do what we can to make these run faster, and one of the biggest factors appears to be the `tier` we use.

# Related PR(s), if any:

* https://github.com/GoogleCloudPlatform/magic-modules/pull/6986

**Comment by roaks3** [comment]
@mmizutani already got this started a bit with https://github.com/GoogleCloudPlatform/magic-modules/pull/6986, specifically [this comment](https://github.com/GoogleCloudPlatform/magic-modules/pull/6986#discussion_r1070563763). These reverted commits contain the changes they were looking at:
- https://github.com/GoogleCloudPlatform/magic-modules/pull/6986/commits/fc399bea2d6f5ff96605feed057f4bb7697d6da0
- https://github.com/GoogleCloudPlatform/magic-modules/pull/6986/commits/c454b9e6e9a2d0bc40004eb6beeb9263e1efe4fc

In short, the `db-1-micro` tier for Postgres appears rather slow when compared to the other options, so it seems like we can get a big speed bump simply by switching off of that. I'm trying to engage with the Cloud SQL team internally to understand what is happening, and if they have any other suggestions around this.

**Comment by roaks3** [comment]
Here are the run times from our nightly tests. I've included all SQL-related tests that run for > 15 mins, sampled before this change, and after.

Apr 16:

* TestAccSqlDatabaseInstance_activationPolicy 55m 15s 890ms
* TestAccSqlDatabaseInstance_cloneWithSettings 45m 15s 875ms
* TestAccSqlDatabaseInstance_basicClone 40m 19s 987ms
* TestAccSqlDatabaseInstance_cloneWithDatabaseNames 38m 59s 609ms
* TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRangeClone 34m 56s 542ms
* TestAccSqlDatabaseInstance_deleteDefaultUserBeforeSubsequentApiCalls 31m 43s 886ms
* TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRangeReplica 27m 50s 894ms
* TestAccSqlDatabaseInstance_backupUpdate 26m 26s 627ms
* TestAccSqlDatabaseInstance_createFromBackup 26m 21s 376ms
* TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled 25m 35s 788ms
* TestAccSqlDatabaseInstance_slave 23m 11s 295ms
* TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRange 20m 50s 277ms
* TestAccSqlDatabaseInstance_withPSCEnabled_withAllowedConsumerProjects 19m 26s 566ms
* TestAccSqlDatabaseInstance_withPSCEnabled_thenAddAllowedConsumerProjects_thenRemoveAllowedConsumerProject 18m 53s 723ms
* TestAccSqlDatabaseInstance_basicInstance_thenPSCEnabled 18m 49s 144ms
* TestAccSqlDatabaseInstance_withPSCEnabled_withEmptyAllowedConsumerProjects 18m 34s 230ms
* TestAccSqlDatabaseInstance_withPSCEnabled_withoutAllowedConsumerProjects 18m 03s 595ms
* TestAccSqlUser_iamUser 17m 34s 264ms
* TestAccSqlDatabaseInstance_settings_upgrade 17m 13s 783ms
* TestAccSqlDatabaseInstance_BackupRetention 17m 06s 699ms
* TestAccSqlDatabaseInstance_replica 16m 11s 727ms
* TestAccSqlDatabaseInstance_Edition_Downgrade 15m 51s 980ms
* TestAccSqlDatabaseInstance_Edition 15m 40s 843ms

June 3:

* TestAccSqlDatabaseInstance_cloneWithSettings 45m 46s 147ms
* TestAccSqlDatabaseInstance_basicClone 40m 01s 833ms
* TestAccSqlDatabaseInstance_cloneWithDatabaseNames 38m 59s 363ms
* TestAccSqlDatabaseInstance_activationPolicy 29m 32s 787ms
* TestAccSqlDatabaseInstance_deleteDefaultUserBeforeSubsequentApiCalls 18m 49s 658ms
* TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRangeClone 17m 50s 484ms
* TestAccDataSourceSqlDatabaseInstances_regionFilter 16m 42s 893ms
* TestAccSqlDatabaseInstance_replica 16m 41s 734ms
* TestAccSqlClientCert_postgres 16m 13s 452ms
* TestAccSqlUser_iamUser 15m 51s 654ms
* TestAccDataSourceSqlDatabaseInstances_basic 15m 47s 300ms
* TestAccSqlUser_mysqlPasswordPolicy 15m 07s 765ms

**Comment by roaks3** [comment]
In short, it looks like the change made the run times much better. The clone-related tests appear to have remained unaffected, so I would wonder if we can apply a similar change to those tests, but otherwise this looks to be resolved.
